common/psbt_open.h

```diff
@@ -0,0 +1,144 @@
+#ifndef LIGHTNING_COMMON_PSBT_OPEN_H
+#define LIGHTNING_COMMON_PSBT_OPEN_H
+#include "config.h"
+#include <ccan/short_types/short_types.h>
+#include <ccan/tal/tal.h>
+#include <stdbool.h>
+#include <wally_psbt.h>
+#include <wally_transaction.h>
+
+struct wally_tx_input;
+struct wally_tx_output;
+struct wally_psbt;
+struct wally_psbt_input;
+struct wally_psbt_output;
+struct wally_unknowns_map;
+
+struct input_set {
+	struct wally_tx_input tx_input;
+	struct wally_psbt_input input;
+};
+
+struct output_set {
+	struct wally_tx_output tx_output;
+	struct wally_psbt_output output;
+};
+
+#define PSBT_TYPE_SERIAL_ID 0x01
+#define PSBT_TYPE_MAX_WITNESS_LEN 0x02
+
+/* psbt_get_serial_id - Returns the serial_id from an unknowns map
+ *
+ * @map - the map to find the serial id entry within
+ * @serial_id - found serial_id
+ *
+ * Returns false if serial_id is not present
+ */
+WARN_UNUSED_RESULT bool psbt_get_serial_id(struct wally_unknowns_map *map, u16 *serial_id);
+
+/* psbt_sort_by_serial_id - Sort PSBT by serial_ids
+ *
+ * MUST have a serial_id on every input/output.
+ *
+ * @psbt - psbt to sort
+ */
+void psbt_sort_by_serial_id(struct wally_psbt *psbt);
+
+/* psbt_has_diff - Returns set of diffs btw orig + new psbt
+ *
+ * Note that diff is based on *serial_ids*. All inputs+outputs
+ * MUST have a serial_id field present before calling this.
+ *
+ * If a serial_id has remained the same, but the underlying data
+ * has changed, this WILL NOT detect it.
+ *
+ * @ctx - allocation context for returned diffsets
+ * @orig - original psbt
+ * @new - updated psbt
+ * @added_ins - inputs added to {new}
+ * @rm_ins - inputs removed from {new}
+ * @added_outs - outputs added to {new}
+ * @rm_outs - outputs removed from {new}
+ *
+ * Returns true if no changes are found.
```

Comment:
No it doesn't!

Also probably needs warning about sets simply referring into orig and new here, not on implementation?